app/src/main/java/intellistart/interviewplanning/model/booking/validation/BookingValidator.kt

```diff
@@ -11,84 +11,94 @@ import intellistart.interviewplanning.model.period.TimeService
+            return Mono.error(BookingException(BookingExceptionProfile.INVALID_SUBJECT))
         }
-        if (description.length > DESCRIPTION_MAX_SIZE) {
-            throw BookingException(BookingExceptionProfile.INVALID_DESCRIPTION)
+        return if (description.length > DESCRIPTION_MAX_SIZE) {
+            Mono.error(BookingException(BookingExceptionProfile.INVALID_DESCRIPTION))
+        } else {
+            Unit.toMono()
         }
     }
 
-    private fun checkBookingForOverlapsWithSlots(booking: Booking) {
-        val periodOfInterviewer: Period = slotService.getById(booking.interviewerSlotId).period
-        val periodOfCandidate: Period = slotService.getById(booking.candidateSlotId).period
-
+    private fun checkBookingForOverlapsWithSlots(booking: Booking): Mono<Unit> = Mono.zip(
+        slotService.getById(booking.interviewerSlotId),
+        slotService.getById(booking.candidateSlotId)
+    ).flatMap { tuple ->
+        val periodOfInterviewer: Period = tuple.t1.period
+        val periodOfCandidate: Period = tuple.t2.period
         val bookingPeriod: Period = booking.period
 
         if (
             !periodOfInterviewer.date.isEqual(periodOfCandidate.date) ||
             !periodService.isFirstInsideSecond(bookingPeriod, periodOfInterviewer) ||
             !periodService.isFirstInsideSecond(bookingPeriod, periodOfCandidate)
         ) {
-            throw BookingException(BookingExceptionProfile.SLOTS_NOT_INTERSECTING)
+            Mono.error(BookingException(BookingExceptionProfile.SLOTS_NOT_INTERSECTING))
+        } else {
+            Unit.toMono()
         }
     }
 
     private fun checkBookingForOverlapsWithOtherBookings(
-        booking: Booking
-    ) {
-        validatePeriodNotOverlappingWithOtherBookingPeriods(
-            booking.id,
-            booking.period,
-            slotService.getById(booking.interviewerSlotId).bookings
-        )
+        booking: Booking,
+    ): Mono<Unit> = Mono.zip(
+        slotService.getById(booking.interviewerSlotId),
+        slotService.getById(booking.candidateSlotId)
+    ).flatMap { tuple ->
+        val interviewerBookings: Collection<Booking> = tuple.t1.bookings
+        val candidateBookings: Collection<Booking> = tuple.t2.bookings
 
         validatePeriodNotOverlappingWithOtherBookingPeriods(
             booking.id,
             booking.period,
-            slotService.getById(booking.candidateSlotId).bookings
+            interviewerBookings
+        ).then(
+            validatePeriodNotOverlappingWithOtherBookingPeriods(
+                booking.id,
+                booking.period,
+                candidateBookings
+            )
         )
     }
 
     private fun validatePeriodNotOverlappingWithOtherBookingPeriods(
         updatingBookingId: ObjectId,
         period: Period,
-        bookings: Collection<Booking>
-    ) {
+        bookings: Collection<Booking>,
+    ): Mono<Unit> {
```

Comment:
This function should not be reactive. Let's revert back to simple `Unit` return type.